index.js

```diff
@@ -20,40 +21,48 @@ module.exports = function (tmpl, format) {
 
 		// Do nothing if no contents
 		if (file.isNull()) {
-			this.push(file);
-			return callback();
+			return callback(null, file);
 		}
 
 		if (file.isStream()) {
-			this.emit('error', new gutil.PluginError('gulp-wrap-js', 'Stream content is not supported'));
-			return callback();
+			return callback(new gutil.PluginError('gulp-wrap-js', 'Stream content is not supported'));
 		}
 
+		var err;
 		// check if file.contents is a `Buffer`
 		if (file.isBuffer()) {
-			var ast = esprima.parse(file.contents, {
-				loc: true,
-				source: file.relative,
-				range: true,
-				tokens: true,
-				comment: true
-			});
-			escodegen.attachComments(ast, ast.comments, ast.tokens);
-			ast = tmpl(ast);
-			var result = escodegen.generate(ast, {
-				comment: true,
-				format: format,
-				sourceMap: true,
-				sourceMapWithCode: true,
-				file: file.relative
-			});
-			file.contents = new Buffer(result.code);
-			if (file.sourceMap) {
-				applySourceMap(file, JSON.parse(result.map.toString()));
+			try {
+				var ast = esprima.parse(file.contents, {
+					loc: true,
+					source: file.relative,
+					range: true,
+					tokens: true,
+					comment: true
+				});
+			} catch(e) {
+				err = e;
 			}
-			this.push(file);
-		}
 
-		return callback();
+			if(err) {
+				// Relative to gulpfile.js filepath with forward slashes
+				file = gutil.colors.magenta(path.relative('.', file.path).split(path.sep).join('/'));
+				callback(new gutil.PluginError('gulp-wrap-js', file + ' ' + err.message))
```

Comment:
Also - put `return` here and move it inside the `catch` block so that you wouldn't need temporary variable and separate `if-else` block at all.